src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -964,14 +964,18 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private async updateKernelInfoOnKernelChange(kernel: nb.IKernel, kernelAlias?: string) {
 		await this.updateKernelInfo(kernel);
 		kernelAlias = this.kernelAliases.find(kernel => this._defaultLanguageInfo?.name === kernel.toLowerCase()) ?? kernelAlias;
+		// In order to change from kernel alias to other kernel, set kernelAlias to undefined in order to update to new kernel language info
+		if (this._selectedKernelDisplayName !== kernelAlias && this._selectedKernelDisplayName) {
+			kernelAlias = undefined;
```

Comment:
What is the kernelAlias in this case?

Reply:
When we are currently using the Kusto kernel, the kernelAlias would be Kusto. Therefore it would use the kernelAlias to updateLanguageInfo instead of the selected kernel.